lib/scan.ts

```diff
@@ -1,44 +1,76 @@
 import * as fs from 'fs';
-import { fromUrl } from 'hosted-git-info';
 import * as path from 'path';
+
+import { Facts, Options, PluginResponse, ScanResult } from './types';
+import { SignatureResult } from './types';
+import { debug } from './debug';
 import { find } from './find';
+import { fromUrl } from 'hosted-git-info';
+import { getSignature } from './signatures';
 import { getTarget } from './git';
-import { hash } from './hash';
-import { debug } from './debug';
-import {
-  ScanResult,
-  Options,
-  Fingerprint,
-  Facts,
-  PluginResponse,
-} from './types';
+
+const processSignatures = async (
```

Comment:
nit: Stylistically I prefer calling things functions when they are functions, rather than const = etc.

Reply:
The idea of proceed with ```const foo = () => ... ``` 
as function was the immutability benefit that it brings to the wild world of JS/TS.
Saying so I understand that `function` clarifies the intention of the code base.
I am fine with us proceeding with `function`

Reply:
Looking at https://github.com/airbnb/javascript#functions--declarations

> Why? Function declarations are hoisted, which means that it’s easy - too easy - to reference the function before it is defined in the file. This harms readability and maintainability. If you find that a function’s definition is large or complex enough that it is interfering with understanding the rest of the file, then perhaps it’s time to extract it to its own module! Don’t forget to explicitly name the expression, regardless of whether or not the name is inferred from the containing variable (which is often the case in modern browsers or when using compilers such as Babel). This eliminates any assumptions made about the Error’s call stack.

I would prefer us to go with const here.

Reply:
You've explicitly pointed out a doc that points out why these are traps though; you've got an async anonymous function bound to a const which is going to mess with error stacks. The context (IMO) is less readable because its a const rather than calling it a function. The "immutability" is something to be careful of. It isn't immutable, it just can't be reassigned, and since its not a primitive you could modify it.

> Why? Function declarations are hoisted, which means that it’s easy - too easy - to reference the function before it is defined in the file.

You could lint against this, see https://eslint.org/docs/rules/no-use-before-define, if this is a concern.

Reply:
I've changed these to function declarations.